packages/merge-styles/src/Stylesheet.ts

```diff
@@ -122,22 +139,43 @@ export class Stylesheet {
     if (!_stylesheet || (_stylesheet._lastStyleElement && _stylesheet._lastStyleElement.ownerDocument !== document)) {
       const fabricConfig = _global?.FabricConfig || {};
 
-      _stylesheet = _global[STYLESHEET_SETTING] = new Stylesheet(fabricConfig.mergeStyles);
+      const stylesheet = new Stylesheet(fabricConfig.mergeStyles, fabricConfig.serializedStylesheet);
+      _stylesheet = stylesheet;
+      _global[STYLESHEET_SETTING] = stylesheet;
     }
 
     return _stylesheet;
   }
 
-  constructor(config?: IStyleSheetConfig) {
+  constructor(config?: IStyleSheetConfig, serializedStylesheet?: ISerializedStylesheet) {
     this._config = {
-      injectionMode: InjectionMode.insertNode,
+      // If there is no document we won't have an element to inject into.
+      injectionMode: typeof document === 'undefined' ? InjectionMode.none : InjectionMode.insertNode,
```

Comment:
This is the closest this PR gets to breaking changes. Ideally I wouldn't change this, but on server we need to be in `InjectionMode.none` and setting it in config is too late because the Stylesheet is a singleton and who knows what could have already requested an instance and did some style registration, forever lost in the abyss.

In a perfect world this would be `InjectionMode.none` by default and users would enable node insertion on client-side once they are ready to consume styles there.

Reply:
This seems like a good change to me! If document is not defined, it makes sense to not try injecting.